src/p_saveg.c

```diff
@@ -2653,6 +2689,21 @@ void P_ArchiveSpecials (void)
         }
     }
 
+  CheckSaveGame(MAXBUTTONS * sizeof(button_t));
+
+  button_t *button = buttonlist;
+  int i = MAXBUTTONS;
+  do
+  {
+    if (button->btimer != 0)
+    {
+      saveg_write8(tc_button);
+      saveg_write_pad();
+      saveg_write_button_t(button);
+    }
+    button++;
+  } while (--i);
```

Comment:
Why `do {} while()` loop and not `for (int i = 0; i < MAXBUTTONS; ++i)`?

Reply:
Yes, right.